type-definitions/ts-tests/map.ts

```diff
@@ -1,631 +1,647 @@
+import { expect, test } from 'tstyche';
 import { Map, List, MapOf } from 'immutable';
 
-{
-  // #constructor
+test('#constructor', () => {
+  expect(Map()).type.toEqual<Map<unknown, unknown>>();
 
-  // $ExpectType Map<unknown, unknown>
-  Map();
+  expect(Map<number, number>()).type.toEqual<Map<number, number>>();
 
-  // $ExpectType Map<number, number>
-  Map<number, number>();
+  expect(Map([[1, 'a']])).type.toEqual<Map<number, string>>();
 
-  // $ExpectType Map<number, string>
-  Map([[1, 'a']]);
+  expect(Map([['a', 'a']])).type.toEqual<Map<string, string>>();
 
-  // $ExpectType Map<string, string>
-  Map([['a', 'a']]);
+  expect(Map(List<[number, string]>([[1, 'a']]))).type.toEqual<
+    Map<number, string>
+  >();
 
-  // $ExpectType Map<number, string>
-  Map(List<[number, string]>([[1, 'a']]));
+  expect(Map({ a: 1 })).type.toEqual<MapOf<{ a: number }>>();
 
-  // $ExpectType MapOf<{ a: number; }>
-  Map({ a: 1 });
+  expect(Map({ a: 1, b: 'b' })).type.toEqual<MapOf<{ a: number; b: string }>>();
 
-  // $ExpectType MapOf<{ a: number; b: string; }>
-  Map({ a: 1, b: 'b' });
+  expect(Map({ a: Map({ b: Map({ c: 3 }) }) })).type.toEqual<
+    MapOf<{ a: MapOf<{ b: MapOf<{ c: number }> }> }>
+  >();
 
-  // $ExpectType MapOf<{ a: MapOf<{ b: MapOf<{ c: number; }>; }>; }>
-  Map({ a: Map({ b: Map({ c: 3 }) }) });
+  expect(Map<{ a: string }>({ a: 1 })).type.toRaiseError();
 
-  // $ExpectError
-  Map<{ a: string }>({ a: 1 });
+  expect(Map<{ a: string }>({ a: 'a', b: 'b' })).type.toRaiseError();
 
-  // $ExpectError
-  Map<{ a: string }>({ a: 'a', b: 'b' });
+  expect(Map(List([List(['a', 'b'])]))).type.toEqual<
+    MapOf<List<List<string>>>
+  >();
 
-  // No longer works in typescript@>=3.9
-  // // $ExpectError - TypeScript does not support Lists as tuples
-  // Map(List([List(['a', 'b'])]));
+  expect(Map<'status', string>({ status: 'paid' })).type.toEqual<
+    Map<'status', string>
+  >();
 
-  // $ExpectType Map<number, number>
-  const numberMap: Map<number, number> = Map();
```

Comment:
Different from the current test, but it might make sense adding (see comment below):

```ts
expect(Map([[1, 'a']])).type.not.toBeAssignableTo<Map<number, number>>();
```